crates/client/rpc/src/lib.rs

```diff
@@ -1149,18 +1172,18 @@ where
             FieldElement::default()
         };
 
+        let starknet_block_hash = BlockHash(starknet_block.header().hash::<H>().into());
+
+        let state_diff = self.get_state_diff(substrate_block_hash, &starknet_block_hash).map_err(|e| {
+            error!("Failed to get state diff. Substrate block hash: {substrate_block_hash}, error: {e}");
+            StarknetRpcApiError::InternalServerError
+        })?;
+
         Ok(StateUpdate {
             block_hash: starknet_block.header().hash::<H>().into(),
             new_root: Felt252Wrapper::from(self.backend.temporary_global_state_root_getter()).into(),
             old_root,
-            state_diff: StateDiff {
-                storage_diffs: Vec::new(),
-                deprecated_declared_classes: Vec::new(),
-                declared_classes: Vec::new(),
-                deployed_contracts: Vec::new(),
-                replaced_classes: Vec::new(),
-                nonces: Vec::new(),
-            },
+            state_diff,
```

Comment:
could you add this field to simulate tx where it is defaulted to none https://github.com/keep-starknet-strange/madara/blob/main/crates/client/rpc/src/trace_api.rs#L257

Reply:
This is a different type of state diff, just the one induced bt the given transaction. Should be dealt with in a different way. 

Best approach is to have a different type of `BlockifierStateAdapter` that stores intermediary changes imo.